src/OpenTelemetry.SemanticConventions/SemanticConventionsAttributes.cs

```diff
@@ -0,0 +1,5448 @@
+
+        /// <summary>
+        /// MUST be calculated as two different counters starting from <c>1</c> one for sent messages and one for received message.
+        /// </summary>
+        /// <remarks>
+        /// This way we guarantee that the values will be consistent between different implementations.
+        /// </remarks>
+        public const string AttributeMessageId = "message.id";
+
+        /// <summary>
+        /// Whether this is a received or sent message.
+        /// </summary>
+        public const string AttributeMessageType = "message.type";
+
+        /// <summary>
+        /// Uncompressed size of the message in bytes.
+        /// </summary>
+        public const string AttributeMessageUncompressedSize = "message.uncompressed_size";
+
+        /// <summary>
+        /// Prefix for 'attributes.faas.common'.
+        /// </summary>
+        public static readonly string PrefixAttributesFaasCommon = "faas";
+
+        /// <summary>
+        /// Prefix for 'peer'.
+        /// </summary>
+        public static readonly string PrefixPeer = "peer";
+
+        /// <summary>
+        /// Prefix for 'event'.
+        /// </summary>
+        public static readonly string PrefixEvent = "event";
+
+        /// <summary>
+        /// Prefix for 'log.record'.
+        /// </summary>
+        public static readonly string PrefixLogRecord = "log.record";
+
+        /// <summary>
+        /// Prefix for 'log-exception'.
+        /// </summary>
+        public static readonly string PrefixLogException = "exception";
+
+        /// <summary>
+        /// Prefix for 'attributes.log'.
+        /// </summary>
+        public static readonly string PrefixAttributesLog = "log";
+
+        /// <summary>
+        /// Prefix for 'attributes.log.file'.
+        /// </summary>
+        public static readonly string PrefixAttributesLogFile = "log.file";
+
+        /// <summary>
+        /// Prefix for 'ios.lifecycle.events'.
+        /// </summary>
+        public static readonly string PrefixIosLifecycleEvents = "ios";
+
+        /// <summary>
+        /// Prefix for 'android.lifecycle.events'.
+        /// </summary>
+        public static readonly string PrefixAndroidLifecycleEvents = "android";
+
+        /// <summary>
+        /// Prefix for 'messaging.attributes.common'.
+        /// </summary>
+        public static readonly string PrefixMessagingAttributesCommon = "messaging";
+
+        /// <summary>
+        /// Prefix for 'aspnetcore'.
+        /// </summary>
+        public static readonly string PrefixAspnetcore = "aspnetcore";
+
+        /// <summary>
+        /// Prefix for 'signalr.common_attributes'.
+        /// </summary>
+        public static readonly string PrefixSignalrCommonAttributes = "signalr";
+
+        /// <summary>
+        /// Prefix for 'metric.jvm.memory.init'.
+        /// </summary>
+        public static readonly string PrefixMetricJvmMemoryInit = "jvm.memory";
```

Comment:
it could make sense to exclude certain namespaces that are not relevant to .NET (like JVM metrics). Here's an example of how it's done in Java (using new per-namespace codegen) - https://github.com/open-telemetry/semantic-conventions-java/pull/54